volume/devicemapper/devicemapper.go

```diff
@@ -455,29 +473,37 @@ func (d *DeviceMapperDriver) Remove(volumeName string) error {
 		return nil
 	}
 	// get the volume
-	v, err := d.newVolume(volumeName)
-	if err != nil {
-		return err
-	}
-	// remove the snapshots
-	glog.V(1).Infof("Removing snapshots from %s", volumeName)
-	snapshots, err := v.Snapshots()
+	v, err := d.loadVolume(volumeName)
 	if err != nil {
-		return err
-	}
-	for _, snapshot := range snapshots {
-		if err := v.RemoveSnapshot(snapshot); err != nil {
-			return err
+		//log the error, but continue trying to remove things
+		glog.Errorf("Error loading volume %s: %s", volumeName, err)
```

Comment:
We could, but I was trying to be consistent with my change, which is to keep trying to remove whatever we can rather than failing at the first error.